### ISSUE TITLE
Bug in CLUBB interface code

### ISSUE THREAD

**Issue by PeterHjortLauritzen** [ESCOMP/CAM#1053]
### What happened?

@klindsay28 found a bug in the CLUBB interface code (clubb_intr.F90) with this call (commited in tag cam6_3_094):
```
2553     call set_wet_to_dry(state1)
```
Our conversion from wet to dry mixing ratios should have been:

```
  do m = 1,pcnst
     if (cnst_type(m).eq.'wet') then
        state1%q(:ncol,:,m) = state1%q(:ncol,:,m)*state1%pdel(:ncol,:)/state1%pdeldry(:ncol,:)
     endif
  end do
```
but the set_wet_to_dry subroutine call executes:

```
  do m = 1,pcnst
     if (cnst_type(m).eq.'dry') then
        state1%q(:ncol,:,m) = state%q(:ncol,:,m)*state1%pdel(:ncol,:)/state1%pdeldry(:ncol,:)
     endif
  end do
```
So, as Keith states, it makes dry tracers "double dry" and does not convert wet tracers to dry (as the name seems to imply!). That also means that the scaling of the tendencies of wet tracers is incorrect; the tracer tendencies (which are already using moist mixing ratios) are converted to moist again."

### What are the steps to reproduce the bug?

run any configuration of CAM that uses CLUBB!

### What CAM tag were you using?

 cam6_3_161

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Will you be addressing this bug yourself?

Yes

### Extra info

Working on PR ...

**Comment by brian-eaton** [comment]
Hi @PeterHjortLauritzen, @klindsay28, @adamrher.

The name of the subroutine `set_wet_to_dry` is misleading.  But the subroutine, which has been in CAM since cam3_1 (2005) is correct.  The implicit assumption here is that the physics package operates on constituents which are dry except for the water constituents (both mass and number concentrations).  As far as I know this is still true; the moist adjustment schemes assume the values of water species are wet mixing ratios.  The purpose of `set_dry_to_wet` and `set_wet_to_dry` is to convert the 'dry' species mixing ratios, which are known by the public `cnst_type` array, from dry to wet, and then back from wet to dry.  Hence those subroutine are designed to provide conversions for **only** the dry species.  The `cnst_type` array, which is used to distinquish the water (wet) from the non-water (dry) species, is set by the `add_cnst` call for each constituent during the register phase of initialization.

This functionality was originally only used by the FV dycore because its advection scheme assumed wet mr for all species.

The change made to `clubb_intr` in cam6_3_094 is incorrect.  The comment indicates that the intention is to convert all species to dry, and as Keith pointed out this is not what `set_wet_to_dry` does.  What is needed instead are new subroutines in `physics_types` which do the conversions for just the water species.  I would suggest using more explicit names, e.g., `set_water_species_mr_to_dry`, and `set_water_species_mr_to_wet`.

**Comment by PeterHjortLauritzen** [comment]
I think we are in agreement and the bug is fixed in https://github.com/ESCOMP/CAM/pull/1054

(except that the conversion is done in `clubb_intr.F90` and not in a separate subroutine in `physics_types`

**Comment by brian-eaton** [comment]
Fixed in PR #1054.
